File: javascript/1_Fizz_Buzz.js
var fs  = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") {
        var input = line.split(/\s+/);
		var i = 0;
        var answer = [];
		for ( i = 1; i <= input[2]; i++ ) {
			if ( i % input[0] === 0 && i % input[1] === 0 ) {
				answer.push('FB');
			}
			else if ( i % input[0] === 0 ) {
				answer.push('F');
			}
			else if ( i % input[1] === 0 ) {
				answer.push('B');
			}
			else {
				answer.push(i);
			}
		}
        console.log( answer.join(' ') );
    }
});

File: ruby/1_Fizz_Buzz.rb
File.open(ARGV[0]).each_line do |line|
	(f,b,x) = line.split(" ").map(&:to_i)
	for i in 1..x
		if i % f == 0 && i % b == 0
			print "FB"
		elsif i % f == 0
			print "F"
		elsif i % b == 0
			print "B"
		else
			print i
		end
		if i < x
			print " "
		end
	end
	print "\n"
end
